irohad/ametsuchi/impl/postgres_block_query.cpp

```diff
@@ -189,27 +191,39 @@ namespace iroha {
     boost::optional<BlockQuery::wTransaction>
     PostgresBlockQuery::getTxByHashSync(
         const shared_model::crypto::Hash &hash) {
-      auto block = getBlockId(hash) | [this](const auto &block_id) {
-        return block_store_.get(block_id);
-      } | [](const auto &bytes) {
-        return shared_model::converters::protobuf::jsonToModel<
-            shared_model::proto::Block>(bytesToString(bytes));
-      };
-      if (not block) {
-        log_->error("error while converting from JSON");
-        return boost::none;
-      }
-
-      boost::optional<PostgresBlockQuery::wTransaction> result;
-      auto it =
-          std::find_if(block->transactions().begin(),
-                       block->transactions().end(),
-                       [&hash](const auto &tx) { return tx.hash() == hash; });
-      if (it != block->transactions().end()) {
-        result = boost::optional<PostgresBlockQuery::wTransaction>(
-            PostgresBlockQuery::wTransaction(clone(*it)));
-      }
-      return result;
+      return getBlockId(hash) |
+          [this](const auto &block_id) {
+            auto result = this->getBlock(block_id);
+            return result.match(
+                [](expected::Value<
+                    std::unique_ptr<shared_model::interface::Block>> &v)
+                    -> boost::optional<
+                        std::unique_ptr<shared_model::interface::Block>> {
+                  return std::move(v.value);
+                },
+                [this](const expected::Error<std::string> &e)
+                    -> boost::optional<
+                        std::unique_ptr<shared_model::interface::Block>> {
+                  log_->error(e.error);
+                  return boost::none;
+                });
+          }
+      | [&hash, this](const auto &block) {
+          boost::optional<PostgresBlockQuery::wTransaction> transaction;
+          auto it = std::find_if(
+              block->transactions().begin(),
+              block->transactions().end(),
+              [&hash](const auto &tx) { return tx.hash() == hash; });
+          if (it != block->transactions().end()) {
+            transaction = PostgresBlockQuery::wTransaction(clone(*it));
```

Comment:
Can't you return the `PostgresBlockQuery::wTransaction(clone(*it))` immediately from here and avoid using the temporary variable `transaction`?